docs/source/enterprise-guide/change-management.mdx

```diff
@@ -0,0 +1,537 @@
+This is especially true for GraphQL schema definitions, which must be
+coordinated between subgraph services and the gateway.
+
+This section covers the most common change management scenarios in a federated
+architecture. We'll consider the proper sequencing of release events, how Apollo
+tooling makes managing changes easier, and when progressive delivery techniques
+like blue-green deploys are useful.
+
+Specifically, we'll examine scenarios that:
+
+- Involve a schema change
+  - [Backward compatible subgraph schema changes](#backward-compatible-subgraph-schema-changes)
+  - [Backward _incompatible_ subgraph schema changes](#backward-incompatible-subgraph-schema-changes)
+- Involve schema changes in multiple subgraphs
+  - [Changes to shared types](#changes-to-shared-types)
+  - [Changes to query planning](#changes-to-query-planning)
+- Don't involve schema changes
+  - [Changes to a data source](#changes-to-a-data-source)
+  - [Changes to subgraph resolvers](#changes-to-subgraph-resolvers)
+  - [Changes to gateway logic/configuration](#changes-to-gateway-logic-or-configuration)
+
+The release guidelines for each scenario should generally apply to all release
+management systems, whether we're deploying to Kubernetes, serverless, or bare
+metal. Some details might change depending on whether we have a manual
+deployment process, a continuous delivery process, or something in between.
+
+## Backward compatible subgraph schema changes
+
+A schema change is backward compatible if it doesn’t affect existing operations.
+Some examples of backward compatible changes:
+
+- Adding a type or field (unused until a client includes the fields in new
+  operations)
+- Adding an optional argument to existing fields, ideally with a default value
+  (the argument will have a null or default value until a client uses it in new
+  operations)
+- Making a nullable field non-nullable (this restricts the set of possible
+  output values)
+- Removing or changing _unused_ schema elements
+
+> This also includes changes to schema metadata like adding or changing a
+> deprecation or description, but these changes aren’t relevant because they
+> don’t affect runtime behavior.
+
+### Release steps
+
+The release steps for backward compatible changes are as follows:
+
+1. In the subgraph code repository, merge a changeset containing both the schema
+   change and the corresponding resolvers to the release branch, triggering the
+   release pipeline.
+2. The release pipeline performs prerelease steps like running tests and schema
+   checks.
+3. The release pipeline builds the deployable artifacts:
+   - The container, jar, or zip of the runtime code
+   - Deployment manifests such as Kubernetes YAML or SAM templates
+   - The subgraph SDL document
+4. The release pipeline then triggers two deployments either in sequence or in
+   parallel. The steps must be atomic—either both succeed or both fail.
```

Comment:
I'm leaning pretty heavily on the word "atomic" here to avoid getting into the non-ideal state you describe. Probably 

But I think I need to dig into the details with you on this — I'm guessing this is related to the note below about *when* you're publishing the schema.

Interestingly, we make some recommendations on the [order of publish/deploy already](https://www.apollographql.com/docs/federation/managed-federation/deployment/#pushing-configuration-updates-safely) ... but we actually suggest swapping the order depending on whether the change is backward-compatible, which seems very weird to me — how do you automate that in a release pipeline??